e2e/__tests__/readInitialOptions.test.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import path = require('path');
+import execa = require('execa');
+import {readInitialOptions} from 'jest-config';
+
+function resolveFixture(...pathSegments: Array<string>) {
+  return path.resolve(__dirname, '..', 'read-initial-options', ...pathSegments);
+}
+
+describe(readInitialOptions, () => {
+  test('should read from the cwd by default', async () => {
+    const configFile = resolveFixture('js-config', 'jest.config.js');
+    const rootDir = resolveFixture('js-config');
+    jest.spyOn(process, 'cwd').mockReturnValue(rootDir);
```

Comment:
You should spawn `node` inside all the fixtures (like you do for the TS and ESM one), not import and run from here

Reply:
Great idea, fixed using a friendly helper 👌